File: Features/state-reissuance/workflows/src/main/java/net/corda/samples/statereissuance/flows/reissuance/AcceptLandReissuanceFlow.java
package net.corda.samples.statereissuance.flows.reissuance;

import co.paralleluniverse.fibers.Suspendable;
import com.r3.corda.lib.reissuance.flows.ReissueStates;
import com.r3.corda.lib.reissuance.states.ReissuanceRequest;
import net.corda.core.contracts.StateAndRef;
import net.corda.core.contracts.StateRef;
import net.corda.core.crypto.SecureHash;
import net.corda.core.flows.FlowException;
import net.corda.core.flows.FlowLogic;
import net.corda.core.flows.InitiatingFlow;
import net.corda.core.flows.StartableByRPC;
import net.corda.core.identity.Party;
import net.corda.samples.statereissuance.flows.ExitLandTitleFlow;

import java.util.Arrays;
import java.util.List;

@InitiatingFlow
@StartableByRPC
public class AcceptLandReissuanceFlow extends FlowLogic<SecureHash> {

    private StateRef stateRef;
    private Party issuer;

    public AcceptLandReissuanceFlow(StateRef stateRef, Party issuer) {
        this.stateRef = stateRef;
        this.issuer = issuer;
    }

    @Override
    @Suspendable
    public SecureHash call() throws FlowException {

        List<StateAndRef<ReissuanceRequest>> reissueRequestStateAndRefs = getServiceHub().getVaultService()
                .queryBy(ReissuanceRequest.class).getStates();

        StateAndRef<ReissuanceRequest> stateAndRef = reissueRequestStateAndRefs.stream().filter(reissuanceRequestStateAndRef -> {
            ReissuanceRequest reissuanceRequest = reissuanceRequestStateAndRef.getState().getData();
            return reissuanceRequest.getStateRefsToReissue().contains(stateRef);
        }).findAny().orElseThrow(() -> new IllegalArgumentException("ReIssuance Request does not exist"));

        SecureHash txHash = subFlow(new ReissueStates<>(stateAndRef, Arrays.asList(issuer)));

        return txHash;
    }
}

File: kotlin-samples/Features/state-reissuance/workflows/src/main/kotlin/net/corda/samples/statereissuance/flows/reissuance/AcceptLandReissuanceFlow.kt
package net.corda.samples.statereissuance.flows.reissuance

import co.paralleluniverse.fibers.Suspendable
import com.r3.corda.lib.reissuance.flows.ReissueStates
import com.r3.corda.lib.reissuance.flows.ReissueStatesResponder
import com.r3.corda.lib.reissuance.states.ReissuanceRequest
import net.corda.core.contracts.ContractState
import net.corda.core.contracts.StateRef
import net.corda.core.crypto.SecureHash
import net.corda.core.flows.*
import net.corda.core.identity.Party
import net.corda.core.transactions.SignedTransaction
import net.corda.core.utilities.ProgressTracker
import java.util.*

@InitiatingFlow
@StartableByRPC
class AcceptLandReissuance(private val issuer: Party,
                           private val stateRef: StateRef) : FlowLogic<SecureHash>() {
    override val progressTracker = ProgressTracker()

    @Suspendable
    override fun call(): SecureHash {

        val reissueRequestStateAndRefs = serviceHub.vaultService.queryBy(ReissuanceRequest::class.java).states

        val stateAndRef = reissueRequestStateAndRefs.stream().filter { it.state.data.stateRefsToReissue.contains(stateRef) }
                .findAny().orElseThrow { IllegalArgumentException("ReIssuance Request does not exist") }

        return subFlow(ReissueStates<ContractState>(stateAndRef, Arrays.asList(issuer)))
    }
}

@InitiatedBy(ReissueStates::class)
class AcceptLandReissuanceResponder(otherSession: FlowSession) : ReissueStatesResponder(otherSession) {

    override fun checkConstraints(stx: SignedTransaction) {

    }
}